src/base/io-funcs-inl.h

```diff
@@ -224,7 +224,7 @@ template<class T> inline void WriteIntegerVector(std::ostream &os, bool binary,
     os << "]\n";
   }
   if (os.fail()) {
-    throw std::runtime_error("Write failure in WriteIntegerVector.");
+    KALDI_ERR << "Write failure in WriteIntegerVector.";
```

Comment:
how does this affect what is printed?  Because these code paths are often reached, and I think the code that catches this exception prints out the message from the runtime_error.

Reply:
Good question. The face-value answer is the code will print a formatted error and a stack trace to the log that it did not output before, and the what() seen at the catch site will be just 'kaldi::KaldiFatalError', without access to the full error message if an `std::runtime_error&` base class is caught. Somewhat messier than currently, I agree. Let me trace these code paths carefully.

On the other hand, in mu mind this is part of an abnormal program flow. Are there scripts that expect more than a dozen of these as part of their normal execution? Is this something we really need to worry about?

Reply:
Just to clarify, why would these code paths be often reached? This is an `os.fail()`, a genuine write error. Broken pipes? Should not happen often with regular files.

Reply:
Oh, just looked at the file alone. This is in WriteBasicType() and friends; their ReadBasicType() counterparts [use KALDI_ERR from the start](https://github.com/kaldi-asr/kaldi/blob/master/src/base/io-funcs-inl.h#L68). I do not think this should be a concern? Or am I really missing something?

Reply:
I think it's OK.  I'll merge, I checked the catch things and they should catch the KaldiFatalError even if they specifically look for std::runtime_error, as it inherits from it.

Reply:
> they should catch the KaldiFatalError even if they specifically look for std::runtime_error, as it inherits from it

Yes, that was my intent--keep the existing code (both Kaldi and user's) compatible with the new exception class.